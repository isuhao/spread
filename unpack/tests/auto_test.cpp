#include "auto.hpp"
#include <iostream>
#include <assert.h>
#include <exception>

using namespace std;
using namespace Unpack;

void test(const std::string &file)
{
  cout << "Unpacking "+file+":\n  ";
  AutoUnpack unp;
  try
    {
      unp.unpack(file, "_outdir3");
    }
  catch(std::exception &e)
    {
      cout << e.what() << endl;
      return;
    }
  cout << "OK\n";
}

int main()
{
  test("test.sh");
  test("doesn't exist");
  test("archives/test.zip");
  return 0;
}
