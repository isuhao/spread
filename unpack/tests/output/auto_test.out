Unpacking test.sh:
  Error unpacking test.sh: Not a known archive type
Unpacking doesn't exist:
  Error unpacking doesn't exist: Cannot open file
Unpacking archives/test.zip:
  OK
